packages/react-dom/src/server/ReactDOMFizzServerNode.js

```diff
@@ -60,6 +63,8 @@ type Options = {
   unstable_externalRuntimeSrc?: string | BootstrapScriptDescriptor,
   importMap?: ImportMap,
   formState?: ReactFormState<any, any> | null,
+  onHeaders?: (headers: HeadersDescriptor) => void,
+  headersLengthHint?: number,
```

Comment:
Hints seems pretty vague. Can't I specify a max? `maxHeadersLength`?

Reply:
It's not a max because I don't want to compute the next headers length before deciding if it is going to be a header. I chose hint b/c I want it to communicate that this number is a target not a hard limit. You'll get a header object "near" this length. Also it's already a little imprecise b/c it's code unit length which can be 1x to 4x in byte length when sent via utf8. My guess is the average english website will have a byte length around close to 1x the headers length but in other character sets the headers are going to be longer

Reply:
I don't think HTTP headers really support utf8 in practice. Traditionally it was ISO-8859-1 but in practice only the ASCII subset should be used. So either we assume that the URLs have already been encoded or we should arguably encode the URLs ourselves otherwise.